libfwupdplugin/fu-device.c

```diff
@@ -1859,6 +1859,21 @@ fu_device_get_physical_id (FuDevice *self)
 void
 fu_device_add_flag (FuDevice *self, FwupdDeviceFlags flag)
 {
+	/* none is not used as an "exported" flag */
+	if (flag == FWUPD_DEVICE_FLAG_NONE)
+		return;
+
+	/* being both a bootloader and requiring a bootloader is invalid */
+	if (flag == FWUPD_DEVICE_FLAG_NONE ||
+	    flag == FWUPD_DEVICE_FLAG_NEEDS_BOOTLOADER) {
```

Comment:
I think that you should do a bitwise check here - in case multiple flags were sent at once.

Reply:
Agreed, thanks.